### PR TITLE
Added variable `skip_provisioners` to skip 'local-exec'

### LINKED ISSUE #258: Allow local-exec to be skipped
Some environments may not wish to have non-IaC scripts (run by provisioners like local-exec) being run in an automated way.

Where possible, these scripts should be skippable. For example: 

- resource "null_resource" "wait_for_cluster".
- resource "null_resource" "delete_default_kube_dns_configmap"

I am happy to contribute. Would adding a new `skip_provisioners` bool (default to false) be an appropriate solution?

### PR DESCRIPTION
**Pull request by paulpalamarchuk**
* Fix #258
 * Add test `simple_regional_skip_local_exec`
 * Remove old upgrading guide from README's

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -108,6 +108,22 @@ Then perform the following commands on the root folder:
...
+
+v2.0.0 is a breaking release. Refer to the
+[Upgrading to v2.0 guide][upgrading-to-v2.0] for details.
+
+## Upgrade to v1.0.0
+
+Version 1.0.0 of this module introduces a breaking change: adding the `disable-legacy-endpoints` metadata field to all node pools. This metadata is required by GKE and [determines whether the `/0.1/` and `/v1beta1/` paths are available in the nodes' metadata server](https://cloud.google.com/kubernetes-engine/docs/how-to/protecting-cluster-metadata#disable-legacy-apis). If your applications do not require access to the node's metadata server, you can leave the default value of `true` provided by the module. If your applications require access to the metadata server, be sure to read the linked documentation to see if you need to set the value for this field to `false` to allow your applications access to the above metadata server paths.
+
+In either case, upgrading to module version `v1.0.0` will trigger a recreation of all node pools in the cluster.
+
```
Comment (kopachevsky) [inline README.md]: why it's showing up as new change?

File: autogen/dns.tf [hunk 2]
```diff
@@ -20,7 +20,7 @@
   Delete default kube-dns configmap
  *****************************************/
 resource "null_resource" "delete_default_kube_dns_configmap" {
-  count = local.custom_kube_dns_config || local.upstream_nameservers_config ? 1 : 0
+  count = (local.custom_kube_dns_config || local.upstream_nameservers_config) || var.skip_provisioners ? 1 : 0
```
Comment (kopachevsky) [inline autogen/dns.tf]: need to add explanation of this in skip_provisioners var description

File: README.md (hunk 1 above)
Comment (paulpalamarchuk) [reply README.md]: Thank for notice.
Removed old upgrading guides all README's. In addition to [commit](https://github.com/terraform-google-modules/terraform-google-kubernetes-engine/commit/fc3fc6aeb97d3a076ed7699c45c05bb692ca8830#diff-04c6e90faac2675aa89e2176d2eec7d8)

File: autogen/dns.tf (hunk 2 above)
Comment (paulpalamarchuk) [reply autogen/dns.tf]: Updated description of the variable.

File: .kitchen.yml [hunk 3]
```diff
@@ -131,3 +131,10 @@ suites:
       systems:
         - name: workload_metadata_config
           backend: local
+  - name: "simple_regional_skip_local_exec"
```
Comment (morgante) [inline .kitchen.yml]: This doesn't need to be a separate suite, just add it to one of the existing ones.

File: autogen/README.md [hunk 4]
```diff
@@ -122,22 +122,6 @@ Then perform the following commands on the root folder:
 - `terraform apply` to apply the infrastructure build
 - `terraform destroy` to destroy the built infrastructure
 
-## Upgrade to v3.0.0
```
Comment (morgante) [inline autogen/README.md]: Why did you remove this? Please rebase.

File: autogen/cluster.tf [hunk 5]
```diff
@@ -352,6 +352,7 @@ resource "google_container_node_pool" "pools" {
 }
 
 resource "null_resource" "wait_for_cluster" {
+  count = var.skip_provisioners ? 1 : 0
```
Comment (morgante) [inline autogen/cluster.tf]: If you skip provisioners, shouldn't this be count of 0? Please pay attention to quality of work and correct logic.

File: autogen/dns.tf (hunk 2 above)
Comment (morgante) [reply autogen/dns.tf]: This logic is wrong.

If you skip provisioners, no local-execs should *ever* be initialized.

File: autogen/README.md [hunk 6]
```diff
@@ -201,6 +201,7 @@ In either case, upgrading to module version `v1.0.0` will trigger a recreation o
 | resource\_usage\_export\_dataset\_id | The dataset id for which network egress metering for this cluster will be enabled. If enabled, a daemonset will be created in the cluster to meter network egress traffic. | string | `""` | no |
 | sandbox\_enabled | (Beta) Enable GKE Sandbox (Do not forget to set `image_type` = `COS_CONTAINERD` and `node_version` = `1.12.7-gke.17` or later to use it). | bool | `"false"` | no |
 | service\_account | The service account to run nodes as if not overridden in `node_pools`. The create_service_account variable default value (true) will cause a cluster-specific service account to be created. | string | `""` | no |
+| skip\_provisioners | Flag to skip local-exec provisioners. Does not affect if `stub_domains` or `upstream_nameservers` variable set. | bool | `"false"` | no |
```
Comment (kopachevsky) [inline autogen/README.md]: change this one

File: autogen/variables.tf [hunk 7]
```diff
@@ -311,6 +311,11 @@ variable "cluster_resource_labels" {
   default     = {}
 }
 
+variable "skip_provisioners" {
+  type        = bool
+  description = "Flag to skip all local-exec provisioners. It breaks down `stub_domains` and `upstream_nameservers` variables functionality."
```
Comment (morgante) [inline autogen/variables.tf]:
```suggestion
  description = "Flag to skip all local-exec provisioners. It breaks `stub_domains` and `upstream_nameservers` variables functionality."
```

Comment (paulpalamarchuk) [reply autogen/variables.tf]: updated

File: autogen/cluster.tf (hunk 5 above)
Comment (paulpalamarchuk) [reply autogen/cluster.tf]: FIxed

File: autogen/dns.tf (hunk 2 above)
Comment (paulpalamarchuk) [reply autogen/dns.tf]: Fixed

File: .kitchen.yml (hunk 3 above)
Comment (paulpalamarchuk) [reply .kitchen.yml]: Moved to `simple_regional`

File: scripts/wait-for-cluster.sh [hunk 8]
```diff
@@ -15,10 +15,6 @@
 
 set -e
 
-if [ -n "${GOOGLE_APPLICATION_CREDENTIALS}" ]; then
```
Comment (morgante) [inline scripts/wait-for-cluster.sh]: Why are you removing this?
